pkg/cache/v3/cache.go

```diff
@@ -37,6 +36,34 @@ type Request = discovery.DiscoveryRequest
 // DeltaRequest is an alias for the delta discovery request type.
 type DeltaRequest = discovery.DeltaDiscoveryRequest
 
+// Subscription stores the server view of the client state for a given resource type.
+// This allows proper implementation of stateful aspects of the protocol (e.g. returning only some updated resources).
+// Though the methods may return mutable parts of the state for performance reasons,
+// the cache is expected to consider this state as immutable and thread safe between a watch creation and its cancellation.
+type Subscription interface {
+	// ReturnedResources returns a list of resources that clients have ACK'd and their associated version.
+	// The versions are:
+	//  - delta protocol: version of the specific resource set in the response
+	//  - sotw protocol: version of the global response when the resource was last ACKed
+	ReturnedResources() map[string]string
```

Comment:
Why not `AckedResources`?

Reply:
We now set the resources when sending the reply. Given the nonce behavior and the lack of support for NACK (handling it without further work would just create an infinite loop), there is no real point in distinguishing them in sotw for now as we explicitly ignore staled requests